### ISSUE TITLE
Specify data types for input data

### ISSUE THREAD

**Issue by herman-nside** [danielegrattarola/spektral#164]
This is not urgent, but might be helpful to others:

For many of the convolutional layers (except for `GraphSage`), the adjacency matrix needs to be specified as type `float`, not type `int`. `nx.adjacency_matrix(g)` returns `int`s. It would be good to mention this in the documentation somewhere. If I have missed this in the documentation, then my apologies. There might be an easier way, but for me what worked was 
```python
a = np.array(nx.adjacency_matrix(g).todense()).astype(float)
```

**Comment by danielegrattarola** [comment]
I might need to have a deeper look at this. 
Your solution is correct (a more compact way of doing this is `a = nx.adjacency_matrix(g).A.astype(float)`) but type compatibility should be checked at the layer level. 

Leaving this open until I have worked on this.

**Comment by danielegrattarola** [comment]
This issue is solved by commit 2f55c15d87bbf752a390834a0af17e009efb097c for most layers.

All Layers based on `Conv` now auto-cast the adjacency matrix from int to float if the node features are float, and most layers now support combinations of int32, int64, float16, float32, and float64 inputs. 

Almost all layers now support int32/64 adjacency matrices and float32/64 node features. 
Support for int-int and float16 is still a bit iffy, especially when using sparse adjacency matrices.

Cheers
